packages/block-library/src/image/view.js

```diff
@@ -381,25 +389,25 @@ const { state, actions, callbacks } = store(
 						// If it reaches the width first, it keeps the width and compute the
 						// height.
 						const referenceHeight = offsetWidth / naturalRatio;
-						ctx.imageButtonTop =
+						imageButtonTop =
 							( offsetHeight - referenceHeight ) / 2 +
 							buttonOffsetTop +
 							16;
-						ctx.imageButtonRight = buttonOffsetRight + 16;
+						imageButtonRight = buttonOffsetRight + 16;
 					} else {
 						// If it reaches the height first, it keeps the height and compute
 						// the width.
 						const referenceWidth = offsetHeight * naturalRatio;
-						ctx.imageButtonTop = buttonOffsetTop + 16;
-						ctx.imageButtonRight =
+						imageButtonTop = buttonOffsetTop + 16;
+						imageButtonRight =
 							( offsetWidth - referenceWidth ) / 2 +
 							buttonOffsetRight +
 							16;
 					}
-				} else {
-					ctx.imageButtonTop = buttonOffsetTop + 16;
-					ctx.imageButtonRight = buttonOffsetRight + 16;
 				}
+
+				state.metadata[ imageId ].imageButtonTop = imageButtonTop;
+				state.metadata[ imageId ].imageButtonRight = imageButtonRight;
```

Comment:
@luisherranz we might have to revert to keep the data in context instead, because it is being used in the template, not anywhere else.

https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/image/index.php#L234-L235

It is breaking the icon placement.

![Uploading image.png…]()

Reply:
I can reproduce this as well.

Reply:
We can use derived state, similar to how we are using in other places:

```js
store('...', {
  state: {
    get imageButtonRight() {
      const { imageId } = getContext();
      return state.metadata[imageId].imageButtonRight;
    },
    get imageButtonTop() ...
  },
});
```

```html
<button
  data-wp-style--right="state.imageButtonRight"
  data-wp-style--top="state.imageButtonTop"
>
  ...
</button>
```

Reply:
Thanks for the suggestion. The issue is fixed now.